### ISSUE TITLE
Server cannot be killed

### ISSUE THREAD

**Issue by alesapin** [ClickHouse/ClickHouse#10309]
Steps to reproduce:
1. Build server in debug-mode (possibly not related).
2. Run test `00474_readonly_settings`.
3. `kill server_pid`.
4. In server logs:
```
2020.04.16 18:41:18.044099 [ 54701 ] {} <Trace> BackgroundSchedulePool: Waiting for threads to finish.
2020.04.16 18:41:18.044998 [ 54701 ] {} <Debug> Application: Shut down storages.
2020.04.16 18:41:18.045052 [ 54701 ] {} <Debug> Application: Destroyed global context.
2020.04.16 18:41:18.045455 [ 54701 ] {} <Information> Application: shutting down
2020.04.16 18:41:18.045492 [ 54701 ] {} <Debug> Application: Uninitializing subsystem: Logging Subsystem
2020.04.16 18:41:18.045595 [ 54704 ] {} <Trace> BaseDaemon: Received signal -2
2020.04.16 18:41:18.045769 [ 54704 ] {} <Information> BaseDaemon: Stop SignalListener thread
```
but  process still alive
5. Thread dump:
```
#0  0x00007f35c7f29d2d in __GI___pthread_timedjoin_ex (threadid=139867608573696, thread_return=0x0, abstime=0x0, block=<optimized out>) at pthread_join_common.c:89
#1  0x00000000261d9b53 in std::__1::__libcpp_thread_join (__t=0x7f3598213350) at /home/alesap/code/cpp/ClickHouse/contrib/libcxx/include/__threading_support:409
#2  0x00000000261d9481 in std::__1::thread::join (this=0x7f3598213350) at /home/alesap/code/cpp/ClickHouse/contrib/libcxx/src/thread.cpp:56
#3  0x00000000154a121b in ThreadPoolImpl<std::__1::thread>::finalize (this=0x30572b98 <GlobalThreadPool::instance()::ret>) at /home/alesap/code/cpp/ClickHouse/src/Common/ThreadPool.cpp:177
#4  0x00000000154a107e in ThreadPoolImpl<std::__1::thread>::~ThreadPoolImpl (this=0x30572b98 <GlobalThreadPool::instance()::ret>) at /home/alesap/code/cpp/ClickHouse/src/Common/ThreadPool.cpp:163
#5  0x00000000154a4638 in GlobalThreadPool::~GlobalThreadPool (this=0x30572b98 <GlobalThreadPool::instance()::ret>) at /home/alesap/code/cpp/ClickHouse/src/Common/ThreadPool.h:129
#6  0x00007f35c8521041 in __run_exit_handlers (status=0, listp=0x7f35c88c9718 <__exit_funcs>, run_list_atexit=run_list_atexit@entry=true, run_dtors=run_dtors@entry=true) at exit.c:108
#7  0x00007f35c852113a in __GI_exit (status=<optimized out>) at exit.c:139
#8  0x00007f35c84ffb9e in __libc_start_main (main=0x15456240 <main(int, char**)>, argc=3, argv=0x7ffe6881b268, init=<optimized out>, fini=<optimized out>, rtld_fini=<optimized out>, stack_end=0x7ffe6881b258) at ../csu/libc-start.c:344
#9  0x000000001545602a in _start ()

Thread 5 (Thread 0x7f35771fe700 (LWP 711)):
#0  0x00007f35c7f2ef85 in futex_abstimed_wait_cancelable (private=<optimized out>, abstime=0x7f35771f8be0, expected=0, futex_word=0x7f35c78b1050) at ../sysdeps/unix/sysv/linux/futex-internal.h:205
#1  __pthread_cond_wait_common (abstime=0x7f35771f8be0, mutex=0x7f35c78b1000, cond=0x7f35c78b1028) at pthread_cond_wait.c:539
#2  __pthread_cond_timedwait (cond=0x7f35c78b1028, mutex=0x7f35c78b1000, abstime=0x7f35771f8be0) at pthread_cond_wait.c:667
#3  0x000000002614094b in std::__1::__libcpp_condvar_timedwait (__cv=0x7f35c78b1028, __m=0x7f35c78b1000, __ts=0x7f35771f8be0) at /home/alesap/code/cpp/ClickHouse/contrib/libcxx/include/__threading_support:359
#4  0x000000002614066f in std::__1::condition_variable::__do_timed_wait (this=0x7f35c78b1028, lk=..., tp=...) at /home/alesap/code/cpp/ClickHouse/contrib/libcxx/src/condition_variable.cpp:74
#5  0x00000000154a7d39 in std::__1::condition_variable::wait_for<long long, std::__1::ratio<1l, 1000000000l> > (this=0x7f35c78b1028, __lk=..., __d=...) at /home/alesap/code/cpp/ClickHouse/contrib/libcxx/include/__mutex_base:468
#6  0x00000000154a783e in std::__1::condition_variable::__do_timed_wait<std::__1::chrono::steady_clock> (this=0x7f35c78b1028, __lk=..., __tp=...) at /home/alesap/code/cpp/ClickHouse/contrib/libcxx/include/__mutex_base:523
#7  0x00000000154a74e7 in std::__1::condition_variable::wait_until<std::__1::chrono::steady_clock, std::__1::chrono::duration<long long, std::__1::ratio<1l, 1000000000l> > > (this=0x7f35c78b1028, __lk=..., __t=...) at /home/alesap/code/cpp/ClickHouse/contrib/libcxx/include/__mutex_base:426
#8  0x000000001e1d84f5 in std::__1::condition_variable::wait_until<std::__1::chrono::steady_clock, std::__1::chrono::duration<long long, std::__1::ratio<1l, 1000000000l> >, DB::NamedSessions::cleanThread()::{lambda()#1}>(std::__1::unique_lock<std::__1::mutex>&, std::__1::chrono::time_point<std::__1::chrono::steady_clock, std::__1::chrono::duration<long long, std::__1::ratio<1l, 1000000000l> > > const&, DB::NamedSessions::cleanThread()::{lambda()#1}) (this=0x7f35c78b1028, __lk=..., __t=..., __pred=...) at /home/alesap/code/cpp/ClickHouse/contrib/libcxx/include/__mutex_base:438
#9  0x000000001e1d7b3d in std::__1::condition_variable::wait_for<long long, std::__1::ratio<1l, 1000000000l>, DB::NamedSessions::cleanThread()::{lambda()#1}>(std::__1::unique_lock<std::__1::mutex>&, std::__1::chrono::duration<long long, std::__1::ratio<1l, 1000000000l> > const&, DB::NamedSessions::cleanThread()::{lambda()#1}) (this=0x7f35c78b1028, __lk=..., __d=..., __pred=...) at /home/alesap/code/cpp/ClickHouse/contrib/libcxx/include/__mutex_base:482
#10 0x000000001e1d6c4f in DB::NamedSessions::cleanThread (this=0x7f35c78b0f90) at /home/alesap/code/cpp/ClickHouse/src/Interpreters/Context.cpp:217
#11 0x000000001e1dac19 in std::__1::__invoke_constexpr<void (DB::NamedSessions::* const&)(), DB::NamedSessions* const&, , void> (__f=@0x7f3576400018: (void (DB::NamedSessions::*)(DB::NamedSessions * const)) 0x1e1d6b90 <DB::NamedSessions::cleanThread()>, __a0=@0x7f3576400028: 0x7f35c78b0f90) at /home/alesap/code/cpp/ClickHouse/contrib/libcxx/include/type_traits:3467
#12 0x000000001e1dab20 in std::__1::__apply_tuple_impl<void (DB::NamedSessions::* const&)(), std::__1::tuple<DB::NamedSessions*> const&, 0ul> (__f=@0x7f3576400018: (void (DB::NamedSessions::*)(DB::NamedSessions * const)) 0x1e1d6b90 <DB::NamedSessions::cleanThread()>, __t=...) at /home/alesap/code/cpp/ClickHouse/contrib/libcxx/include/tuple:1415
#13 0x000000001e1daac8 in std::__1::apply<void (DB::NamedSessions::* const&)(), std::__1::tuple<DB::NamedSessions*> const&> (__f=@0x7f3576400018: (void (DB::NamedSessions::*)(DB::NamedSessions * const)) 0x1e1d6b90 <DB::NamedSessions::cleanThread()>, __t=...) at /home/alesap/code/cpp/ClickHouse/contrib/libcxx/include/tuple:1424
#14 0x000000001e1da98e in ThreadFromGlobalPool::ThreadFromGlobalPool<void (DB::NamedSessions::*)(), DB::NamedSessions*>(void (DB::NamedSessions::*&&)(), DB::NamedSessions*&&)::{lambda()#1}::operator()() const (this=0x7f3576400008) at /home/alesap/code/cpp/ClickHouse/src/Common/ThreadPool.h:160
#15 0x000000001e1da8f3 in std::__1::__invoke<ThreadFromGlobalPool::ThreadFromGlobalPool<void (DB::NamedSessions::*)(), DB::NamedSessions*>(void (DB::NamedSessions::*&&)(), DB::NamedSessions*&&)::{lambda()#1}&> (__f=...) at /home/alesap/code/cpp/ClickHouse/contrib/libcxx/include/type_traits:3519
#16 0x000000001e1da873 in std::__1::__invoke_void_return_wrapper<void>::__call<ThreadFromGlobalPool::ThreadFromGlobalPool<void (DB::NamedSessions::*)(), DB::NamedSessions*>(void (DB::NamedSessions::*&&)(), DB::NamedSessions*&&)::{lambda()#1}&>(ThreadFromGlobalPool::ThreadFromGlobalPool<void (DB::NamedSessions::*)(), DB::NamedSessions*>(void (DB::NamedSessions::*&&)(), DB::NamedSessions*&&)::{lambda()#1}&) (__args=...) at /home/alesap/code/cpp/ClickHouse/contrib/libcxx/include/__functional_base:348
#17 0x000000001e1da833 in std::__1::__function::__alloc_func<ThreadFromGlobalPool::ThreadFromGlobalPool<void (DB::NamedSessions::*)(), DB::NamedSessions*>(void (DB::NamedSessions::*&&)(), DB::NamedSessions*&&)::{lambda()#1}, std::__1::allocator<{lambda()#1}>, void ()>::operator()() (this=0x7f3576400008) at /home/alesap/code/cpp/ClickHouse/contrib/libcxx/include/functional:1540
#18 0x000000001e1d9345 in std::__1::__function::__func<ThreadFromGlobalPool::ThreadFromGlobalPool<void (DB::NamedSessions::*)(), DB::NamedSessions*>(void (DB::NamedSessions::*&&)(), DB::NamedSessions*&&)::{lambda()#1}, std::__1::allocator<{lambda()#1}>, void ()>::operator()() (this=0x7f3576400000) at /home/alesap/code/cpp/ClickHouse/contrib/libcxx/include/functional:1714
#19 0x000000001548e832 in std::__1::__function::__value_func<void ()>::operator()() const (this=0x7f35771f9470) at /home/alesap/code/cpp/ClickHouse/contrib/libcxx/include/functional:1867
#20 0x000000001548c6cb in std::__1::function<void ()>::operator()() const (this=0x7f35771f9470) at /home/alesap/code/cpp/ClickHouse/contrib/libcxx/include/functional:2473
#21 0x00000000154a16f6 in ThreadPoolImpl<std::__1::thread>::worker (this=0x30572b98 <GlobalThreadPool::instance()::ret>, thread_it=...) at /home/alesap/code/cpp/ClickHouse/src/Common/ThreadPool.cpp:224
#22 0x00000000154abb2b in void ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, int, std::__1::optional<unsigned long>)::{lambda()#3}::operator()() const (this=0x7f3598213368) at /home/alesap/code/cpp/ClickHouse/src/Common/ThreadPool.cpp:105
#23 0x00000000154aba73 in std::__1::__invoke<void ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, int, std::__1::optional<unsigned long>)::{lambda()#3}> (__f=...) at /home/alesap/code/cpp/ClickHouse/contrib/libcxx/include/type_traits:3519
#24 0x00000000154ab8db in std::__1::__thread_execute<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, int, std::__1::optional<unsigned long>)::{lambda()#3}>(std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, int, std::__1::optional<unsigned long>)::{lambda()#3}>&, std::__1::__tuple_indices<>) (__t=...) at /home/alesap/code/cpp/ClickHouse/contrib/libcxx/include/thread:273
#25 0x00000000154aaf08 in std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, int, std::__1::optional<unsigned long>)::{lambda()#3}> >(std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, int, std::__1::optional<unsigned long>)::{lambda()#3}>) (__vp=0x7f3598213360) at /home/alesap/code/cpp/ClickHouse/contrib/libcxx/include/thread:284
#26 0x00007f35c7f286db in start_thread (arg=0x7f35771fe700) at pthread_create.c:463
#27 0x00007f35c85ff88f in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 4 (Thread 0x7f35af5ff700 (LWP 627)):
#0  0x00007f35c7f2eed9 in futex_reltimed_wait_cancelable (private=<optimized out>, reltime=0x7f35af5f9b90, expected=0, futex_word=0x7f356b039170) at ../sysdeps/unix/sysv/linux/futex-internal.h:142
#1  __pthread_cond_wait_common (abstime=0x7f35af5f9cb8, mutex=0x7f356b039120, cond=0x7f356b039148) at pthread_cond_wait.c:533
#2  __pthread_cond_timedwait (cond=0x7f356b039148, mutex=0x7f356b039120, abstime=0x7f35af5f9cb8) at pthread_cond_wait.c:667
#3  0x0000000024b33d1a in Poco::SemaphoreImpl::waitImpl (this=0x7f356b039118, milliseconds=10000) at /home/alesap/code/cpp/ClickHouse/contrib/poco/Foundation/src/Semaphore_POSIX.cpp:157
#4  0x0000000015612fe7 in Poco::Semaphore::tryWait (this=0x7f356b039118, milliseconds=10000) at /home/alesap/code/cpp/ClickHouse/contrib/poco/Foundation/include/Poco/Semaphore.h:130
#5  0x000000001fecd979 in ConcurrentBoundedQueue<Coordination::TestKeeper::RequestInfo>::tryPop (this=0x7f356b0390c0, x=..., milliseconds=10000) at /home/alesap/code/cpp/ClickHouse/src/Common/ConcurrentBoundedQueue.h:122
#6  0x000000001feb4e48 in Coordination::TestKeeper::processingThread (this=0x7f356b039000) at /home/alesap/code/cpp/ClickHouse/src/Common/ZooKeeper/TestKeeper.cpp:462
#7  0x000000001feb932e in Coordination::TestKeeper::TestKeeper(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, Poco::Timespan)::$_0::operator()() const (this=0x7f35ae800018) at /home/alesap/code/cpp/ClickHouse/src/Common/ZooKeeper/TestKeeper.cpp:432
#8  0x000000001feb92f3 in _ZNSt3__118__invoke_constexprIRKZN12Coordination10TestKeeperC1ERKNS_12basic_stringIcNS_11char_traitsIcEENS_9allocatorIcEEEEN4Poco8TimespanEE3$_0JEEEDTclclsr3std3__1E7forwardIT_Efp_Espclsr3std3__1E7forwardIT0_Efp0_EEEOSG_DpOSH_ (__f=...) at /home/alesap/code/cpp/ClickHouse/contrib/libcxx/include/type_traits:3525
#9  0x000000001feb9287 in _ZNSt3__118__apply_tuple_implIRKZN12Coordination10TestKeeperC1ERKNS_12basic_stringIcNS_11char_traitsIcEENS_9allocatorIcEEEEN4Poco8TimespanEE3$_0RKNS_5tupleIJEEEJEEEDcOT_OT0_NS_15__tuple_indicesIJXspT1_EEEE (__f=..., __t=...) at /home/alesap/code/cpp/ClickHouse/contrib/libcxx/include/tuple:1415
#10 0x000000001feb9248 in std::__1::apply<Coordination::TestKeeper::TestKeeper(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, Poco::Timespan)::$_0(std::__1::tuple<> const&) const&> (__f=..., __t=...) at /home/alesap/code/cpp/ClickHouse/contrib/libcxx/include/tuple:1424
#11 0x000000001feb910e in _ZZN20ThreadFromGlobalPoolC1IZN12Coordination10TestKeeperC1ERKNSt3__112basic_stringIcNS3_11char_traitsIcEENS3_9allocatorIcEEEEN4Poco8TimespanEE3$_0JEEEOT_DpOT0_ENKUlvE_clEv (this=0x7f35ae800008) at /home/alesap/code/cpp/ClickHouse/src/Common/ThreadPool.h:160
#12 0x000000001feb9073 in _ZNSt3__18__invokeIRZN20ThreadFromGlobalPoolC1IZN12Coordination10TestKeeperC1ERKNS_12basic_stringIcNS_11char_traitsIcEENS_9allocatorIcEEEEN4Poco8TimespanEE3$_0JEEEOT_DpOT0_EUlvE_JEEEDTclclsr3std3__1E7forwardISG_Efp_Espclsr3std3__1E7forwardISI_Efp0_EEESH_SK_ (__f=...) at /home/alesap/code/cpp/ClickHouse/contrib/libcxx/include/type_traits:3519
#13 0x000000001feb8ff3 in _ZNSt3__128__invoke_void_return_wrapperIvE6__callIJRZN20ThreadFromGlobalPoolC1IZN12Coordination10TestKeeperC1ERKNS_12basic_stringIcNS_11char_traitsIcEENS_9allocatorIcEEEEN4Poco8TimespanEE3$_0JEEEOT_DpOT0_EUlvE_EEEvDpOT_ (__args=...) at /home/alesap/code/cpp/ClickHouse/contrib/libcxx/include/__functional_base:348
#14 0x000000001feb8fb3 in _ZNSt3__110__function12__alloc_funcIZN20ThreadFromGlobalPoolC1IZN12Coordination10TestKeeperC1ERKNS_12basic_stringIcNS_11char_traitsIcEENS_9allocatorIcEEEEN4Poco8TimespanEE3$_0JEEEOT_DpOT0_EUlvE_NS9_ISM_EEFvvEEclEv (this=0x7f35ae800008) at /home/alesap/code/cpp/ClickHouse/contrib/libcxx/include/functional:1540
#15 0x000000001feb7ae5 in _ZNSt3__110__function6__funcIZN20ThreadFromGlobalPoolC1IZN12Coordination10TestKeeperC1ERKNS_12basic_stringIcNS_11char_traitsIcEENS_9allocatorIcEEEEN4Poco8TimespanEE3$_0JEEEOT_DpOT0_EUlvE_NS9_ISM_EEFvvEEclEv (this=0x7f35ae800000) at /home/alesap/code/cpp/ClickHouse/contrib/libcxx/include/functional:1714
#16 0x000000001548e832 in std::__1::__function::__value_func<void ()>::operator()() const (this=0x7f35af5fa470) at /home/alesap/code/cpp/ClickHouse/contrib/libcxx/include/functional:1867
#17 0x000000001548c6cb in std::__1::function<void ()>::operator()() const (this=0x7f35af5fa470) at /home/alesap/code/cpp/ClickHouse/contrib/libcxx/include/functional:2473
#18 0x00000000154a16f6 in ThreadPoolImpl<std::__1::thread>::worker (this=0x30572b98 <GlobalThreadPool::instance()::ret>, thread_it=...) at /home/alesap/code/cpp/ClickHouse/src/Common/ThreadPool.cpp:224
#19 0x00000000154abb2b in void ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, int, std::__1::optional<unsigned long>)::{lambda()#3}::operator()() const (this=0x7f35c79c8ba8) at /home/alesap/code/cpp/ClickHouse/src/Common/ThreadPool.cpp:105
#20 0x00000000154aba73 in std::__1::__invoke<void ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, int, std::__1::optional<unsigned long>)::{lambda()#3}> (__f=...) at /home/alesap/code/cpp/ClickHouse/contrib/libcxx/include/type_traits:3519
#21 0x00000000154ab8db in std::__1::__thread_execute<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, int, std::__1::optional<unsigned long>)::{lambda()#3}>(std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, int, std::__1::optional<unsigned long>)::{lambda()#3}>&, std::__1::__tuple_indices<>) (__t=...) at /home/alesap/code/cpp/ClickHouse/contrib/libcxx/include/thread:273
#22 0x00000000154aaf08 in std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, int, std::__1::optional<unsigned long>)::{lambda()#3}> >(std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, int, std::__1::optional<unsigned long>)::{lambda()#3}>) (__vp=0x7f35c79c8ba0) at /home/alesap/code/cpp/ClickHouse/contrib/libcxx/include/thread:284
#23 0x00007f35c7f286db in start_thread (arg=0x7f35af5ff700) at pthread_create.c:463
#24 0x00007f35c85ff88f in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 3 (Thread 0x7f35c534f700 (LWP 601)):
#0  0x00007f35c7f2e9f3 in futex_wait_cancelable (private=<optimized out>, expected=0, futex_word=0x7f35c78578e0) at ../sysdeps/unix/sysv/linux/futex-internal.h:88
#1  __pthread_cond_wait_common (abstime=0x0, mutex=0x7f35c7857890, cond=0x7f35c78578b8) at pthread_cond_wait.c:502
#2  __pthread_cond_wait (cond=0x7f35c78578b8, mutex=0x7f35c7857890) at pthread_cond_wait.c:655
#3  0x0000000024aa2cb3 in Poco::EventImpl::waitImpl (this=0x7f35c7857888) at /home/alesap/code/cpp/ClickHouse/contrib/poco/Foundation/src/Event_POSIX.cpp:106
#4  0x0000000015495d8f in Poco::Event::wait (this=0x7f35c7857888) at /home/alesap/code/cpp/ClickHouse/contrib/poco/Foundation/include/Poco/Event.h:97
#5  0x0000000024b4fdc5 in Poco::PooledThread::run (this=0x7f35c7857780) at /home/alesap/code/cpp/ClickHouse/contrib/poco/Foundation/src/ThreadPool.cpp:191
#6  0x0000000024b4b324 in Poco::(anonymous namespace)::RunnableHolder::run (this=0x7f35c7822120) at /home/alesap/code/cpp/ClickHouse/contrib/poco/Foundation/src/Thread.cpp:55
#7  0x0000000024b49daa in Poco::ThreadImpl::runnableEntry (pThread=0x7f35c78577b8) at /home/alesap/code/cpp/ClickHouse/contrib/poco/Foundation/src/Thread_POSIX.cpp:345
#8  0x00007f35c7f286db in start_thread (arg=0x7f35c534f700) at pthread_create.c:463
#9  0x00007f35c85ff88f in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 2 (Thread 0x7f35c5b50700 (LWP 600)):
#0  0x00007f35c7f2e9f3 in futex_wait_cancelable (private=<optimized out>, expected=0, futex_word=0x7f35c7857660) at ../sysdeps/unix/sysv/linux/futex-internal.h:88
#1  __pthread_cond_wait_common (abstime=0x0, mutex=0x7f35c7857610, cond=0x7f35c7857638) at pthread_cond_wait.c:502
#2  __pthread_cond_wait (cond=0x7f35c7857638, mutex=0x7f35c7857610) at pthread_cond_wait.c:655
#3  0x0000000024aa2cb3 in Poco::EventImpl::waitImpl (this=0x7f35c7857608) at /home/alesap/code/cpp/ClickHouse/contrib/poco/Foundation/src/Event_POSIX.cpp:106
#4  0x0000000015495d8f in Poco::Event::wait (this=0x7f35c7857608) at /home/alesap/code/cpp/ClickHouse/contrib/poco/Foundation/include/Poco/Event.h:97
#5  0x0000000024b4fdc5 in Poco::PooledThread::run (this=0x7f35c7857500) at /home/alesap/code/cpp/ClickHouse/contrib/poco/Foundation/src/ThreadPool.cpp:191
#6  0x0000000024b4b324 in Poco::(anonymous namespace)::RunnableHolder::run (this=0x7f35c7822100) at /home/alesap/code/cpp/ClickHouse/contrib/poco/Foundation/src/Thread.cpp:55
#7  0x0000000024b49daa in Poco::ThreadImpl::runnableEntry (pThread=0x7f35c7857538) at /home/alesap/code/cpp/ClickHouse/contrib/poco/Foundation/src/Thread_POSIX.cpp:345
#8  0x00007f35c7f286db in start_thread (arg=0x7f35c5b50700) at pthread_create.c:463
#9  0x00007f35c85ff88f in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 1 (Thread 0x7f35c8ce50c0 (LWP 599)):
#0  0x00007f35c7f29d2d in __GI___pthread_timedjoin_ex (threadid=139867608573696, thread_return=0x0, abstime=0x0, block=<optimized out>) at pthread_join_common.c:89
#1  0x00000000261d9b53 in std::__1::__libcpp_thread_join (__t=0x7f3598213350) at /home/alesap/code/cpp/ClickHouse/contrib/libcxx/include/__threading_support:409
#2  0x00000000261d9481 in std::__1::thread::join (this=0x7f3598213350) at /home/alesap/code/cpp/ClickHouse/contrib/libcxx/src/thread.cpp:56
#3  0x00000000154a121b in ThreadPoolImpl<std::__1::thread>::finalize (this=0x30572b98 <GlobalThreadPool::instance()::ret>) at /home/alesap/code/cpp/ClickHouse/src/Common/ThreadPool.cpp:177
#4  0x00000000154a107e in ThreadPoolImpl<std::__1::thread>::~ThreadPoolImpl (this=0x30572b98 <GlobalThreadPool::instance()::ret>) at /home/alesap/code/cpp/ClickHouse/src/Common/ThreadPool.cpp:163
#5  0x00000000154a4638 in GlobalThreadPool::~GlobalThreadPool (this=0x30572b98 <GlobalThreadPool::instance()::ret>) at /home/alesap/code/cpp/ClickHouse/src/Common/ThreadPool.h:129
#6  0x00007f35c8521041 in __run_exit_handlers (status=0, listp=0x7f35c88c9718 <__exit_funcs>, run_list_atexit=run_list_atexit@entry=true, run_dtors=run_dtors@entry=true) at exit.c:108
#7  0x00007f35c852113a in __GI_exit (status=<optimized out>) at exit.c:139
#8  0x00007f35c84ffb9e in __libc_start_main (main=0x15456240 <main(int, char**)>, argc=3, argv=0x7ffe6881b268, init=<optimized out>, fini=<optimized out>, rtld_fini=<optimized out>, stack_end=0x7ffe6881b258) at ../csu/libc-start.c:344
#9  0x000000001545602a in _start ()
quit
Detaching from program: /home/alesap/code/cpp/BuildCH/programs/clickhouse, process 599
#0  0x00007f44b021fd2d in __GI___pthread_timedjoin_ex (threadid=139931236157184, thread_return=0x0, abstime=0x0, block=<optimized out>) at pthread_join_common.c:89
#1  0x00000000261d9b53 in std::__1::__libcpp_thread_join (__t=0x7f44a5213370) at /home/alesap/code/cpp/ClickHouse/contrib/libcxx/include/__threading_support:409
#2  0x00000000261d9481 in std::__1::thread::join (this=0x7f44a5213370) at /home/alesap/code/cpp/ClickHouse/contrib/libcxx/src/thread.cpp:56
#3  0x00000000154a121b in ThreadPoolImpl<std::__1::thread>::finalize (this=0x30572b98 <GlobalThreadPool::instance()::ret>) at /home/alesap/code/cpp/ClickHouse/src/Common/ThreadPool.cpp:177
#4  0x00000000154a107e in ThreadPoolImpl<std::__1::thread>::~ThreadPoolImpl (this=0x30572b98 <GlobalThreadPool::instance()::ret>) at /home/alesap/code/cpp/ClickHouse/src/Common/ThreadPool.cpp:163
#5  0x00000000154a4638 in GlobalThreadPool::~GlobalThreadPool (this=0x30572b98 <GlobalThreadPool::instance()::ret>) at /home/alesap/code/cpp/ClickHouse/src/Common/ThreadPool.h:129
#6  0x00007f44b0817041 in __run_exit_handlers (status=0, listp=0x7f44b0bbf718 <__exit_funcs>, run_list_atexit=run_list_atexit@entry=true, run_dtors=run_dtors@entry=true) at exit.c:108
#7  0x00007f44b081713a in __GI_exit (status=<optimized out>) at exit.c:139
#8  0x00007f44b07f5b9e in __libc_start_main (main=0x15456240 <main(int, char**)>, argc=3, argv=0x7ffd0c7c7258, init=<optimized out>, fini=<optimized out>, rtld_fini=<optimized out>, stack_end=0x7ffd0c7c7248) at ../csu/libc-start.c:344
#9  0x000000001545602a in _start ()

```

**Comment by alesapin** [comment]
Introduced here https://github.com/ClickHouse/ClickHouse/pull/9232

**Comment by alexey-milovidov** [comment]
Fixed here #10308

**Comment by alexey-milovidov** [comment]
I have checked it manually just in case...
